agent/consul/acl.go

```diff
@@ -1169,7 +1170,30 @@ func (r *ACLResolver) ResolveTokenAndDefaultMeta(token string, entMeta *structs.
 	// Use the meta to fill in the ACL authorization context
 	entMeta.FillAuthzContext(authzContext)
 
-	return result, err
+	return result, nil
+}
+
+type ACLRequest interface {
+	TokenSecret() string
+
+	// Merge sets the fields in this request to the values from source, if the
+	// request does not already have a value.
+	// TODO: rename
+	Merge(source *structs.EnterpriseMeta)
+}
```

Comment:
Almost every one of our existing request structs already implements this interface (as described in #11690). Some may not have a `Merge` method if they don't have EnterpriseMeta, but we can easily implement a no-op version to embed in those structs.